### PR TITLE
Document the scheduler changes for monorepo in cocoon

### PR DESCRIPTION
**Pull request by jtmcdole**
Documentation for presubmits

### CODE REVIEW

**Comment by jtmcdole** [comment]
For the "we get notified of check status" - we are already notified of completion from both LUCI and GitHub. To query the check status; there's already some code in the gitub utility file that could be reused; or we could use a document in firebase that has a TTL on it.

I'm leaning:
1. Keep ci yaml validation open longer; it retains the "block submit"
2. Make a new non-blocking "engine sentitle" for clarity; complete it with engine build status.
3. Finish ci yaml once everything has been scheduled.

@yjbanov - We could remove the blocker on CI Task as well if we use CI yaml validation as the blocker. That leaves one blocker with multiple exists, but a good crumb trail.

Thoughts?

File: app_dart/docs/fusion_schduler.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+What can be noted here is that anyone with commit access can bypass presubmit
+checks as soon as `ci.yaml validation` completes succesfully - and before any
+actual testing completes. The usage of the `autosubmit` label will wait for all
+checks to pass before merging.
+
+At a very high level, presubmits looked like:
+
+```mermaid
+sequenceDiagram
+github --) +cocoon: pr openned
```
Comment (christopherfujino) [inline app_dart/docs/fusion_schduler.md:22]:
```suggestion
github --) +cocoon: pr opened
```

File: app_dart/docs/fusion_schduler.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+deactivate cocoon
+```
+
+### Monorepo Scheduling
+
+There will be multiple `.ci.yaml` files in the repository now that the engine is
+present. Pull requests can contain framework, engine, or both changes. In this
+world: if any parts of the engine are changed, we need to schedule
+**and wait for** and engine builds to be present to test from source. This means
+either `ci.yaml validation` is kept from completing; or anotther check is
```
Comment (christopherfujino) [inline app_dart/docs/fusion_schduler.md:52]:
```suggestion
either `ci.yaml validation` is kept from completing; or another check is
```

File: app_dart/docs/fusion_schduler.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+end
+deactivate cocoon
+```
+
+### Monorepo Scheduling
+
+There will be multiple `.ci.yaml` files in the repository now that the engine is
+present. Pull requests can contain framework, engine, or both changes. In this
+world: if any parts of the engine are changed, we need to schedule
+**and wait for** and engine builds to be present to test from source. This means
```
Comment (christopherfujino) [inline app_dart/docs/fusion_schduler.md:51]:
```suggestion
**and wait for** all engine builds to be present to test from source. This means
```

**Review by yjbanov — APPROVED** [review]
lgtm

**Pull request merged**